p2p/protocol/identify/id.go

```diff
@@ -382,10 +384,23 @@ func (ids *idService) identifyConn(c network.Conn, signal chan struct{}) {
 		return
 	}
 
+	// this is not set I believe, so we have to do it manually
+	if s.Scope().(network.StreamManagementScope).SetProtocol(ID); err != nil {
+		log.Warnf("error setting identify protocol scope: %s", err)
+		s.Reset()
+		return
+	}
+
```

Comment:
@marten-seemann is this assumption correct? Or does the `SetProtocol` above also set it in the scope?
If that's the case, we should remove this code. In fact, we should make it do so and not need this code.

Reply:
It's not implemented yet, but services won't need to call `SetProtocol`. This will be done by the host's stream handler.

Reply:
yeah, identify is just special -- it opens a raw stream and does stuff.

Ok, let's keep it for now, and when it gets implemented we remove it.

Reply:
this is fixed in #1289.

Reply:
fixed in #1289